scalameta/semantic/shared/src/main/scala/scala/meta/semantic/Database.scala

```diff
@@ -1,186 +1,39 @@
+  def structure: String = {
+    val s_entries = entries.map{ case (input, attrs) => s"${input.structure} -> ${attrs.structure}" }.mkString(", ")
+    s"Database(Seq($s_entries))"
   }
+  override def toString: String = syntax
 }
 
-@data class AttributedSource(
-  path: AbsolutePath,
-  names: Map[Anchor, Symbol],
-  messages: Seq[Message],
-  denotations: Map[Symbol, Denotation],
-  sugars: Map[Anchor, String]
-) {
-  override def toString: String = {
-    val lines = mutable.ListBuffer[String]()
-    def appendSection(name: String, section: List[String]): Unit = {
-      if (section.nonEmpty) {
-        lines += (name + ":")
-        lines ++= section
-        lines += ""
-      }
-    }
-
-    lines.append(path.toString, "-" * path.toString.length, "")
-
-    val content = path.slurp
-    val s_names = names.toList.sortBy(_._1.start).map {
-      case ((Anchor(_, start, end), symbol)) =>
-        val snippet = content.substring(start, end)
-        s"[$start..$end): $snippet => $symbol"
-    }
-    appendSection("Names", s_names)
-
-    val s_messages = messages.toList.sortBy(_.anchor.start).map {
-      case Message(Anchor(_, start, end), severity, message) =>
-        val snippet = content.substring(start, end)
-        s"[$start..$end): [${severity.toString.toLowerCase}] ${message}"
-    }
-    appendSection("Messages", s_messages)
-
-    val s_denotations = denotations.toList.sortBy(_._1.syntax).map {
-      case ((symbol, denotation)) =>
-        s"$symbol => $denotation"
-    }
-    appendSection("Denotations", s_denotations)
-
-    val s_sugars = sugars.toList.sortBy(_._1.start).map {
-      case ((Anchor(_, start, end), syntax)) =>
-        s"[$start..$end) $syntax"
-    }
-    appendSection("Sugars", s_sugars)
-
-    lines.mkString(EOL)
-  }
-
-  def toBinary: Array[Byte] = this.toProto[p.AttributedSource].toByteArray
-
-  def writeToFile(file: File): Unit = {
-    file.getParentFile.mkdirs()
-    val fos = new FileOutputStream(file)
-    try fos.write(this.toProto[p.AttributedSource].toByteArray)
-    finally { fos.close() }
-  }
-
-  private[meta] def locateInDatabase(databaseRoot: URI): URI = {
-    var result = databaseRoot.toString
-    if (!result.endsWith("/")) result += "/"
-    // TODO: This relativization is unsound and should be removed.
-    result += path.toRelative.toString
-    result = result.stripSuffix(".scala") + ".semanticdb"
-    new URI(result)
-  }
-}
-
-object AttributedSource {
-  def fromBinary(bytes: Array[Byte]): AttributedSource = {
-    p.AttributedSource.parseFrom(bytes).toMeta[AttributedSource]
-  }
-
-  def fromFile(file: File): AttributedSource = {
-    p.AttributedSource.parseFrom(new FileInputStream(file)).toMeta[AttributedSource]
-  }
-
-  def fromURI(uri: URI): AttributedSource = {
-    p.AttributedSource.parseFrom(uri.toURL.openStream).toMeta[AttributedSource]
+object Database {
+  def load(classpath: Classpath): Database = {
```

Comment:
For #777, this method will need to take a `sourcepath`, because `toSchema` will now take a sourcepath.